### DISCUSSION TITLE
Azure Standard Logic App Service bus Built-in connector - Complete message in Queue action got failed

### DISCUSSION THREAD

**Discussion by Varaprasad424** [Azure/logicapps#836] [Q&A] [answered]
Hi Team, 
In the Logic app standard workflow we are using built-in service bus connector with peek lock trigger. Read message and send message to the queue actions are working fine butwhen we are completing the message in the queue in the logic app workflow we are getting below error. Error :

{

  "code": "ServiceProviderActionFailed",

  "message": "The service provider action failed with error code 'VNetPrivatePortsNotConfigured' and error message 'Using this service provider action in stateful workflows requires VNet integration and the site configuration vnetPrivatePortsCount to be set to a value greater than or equal to 2.'."

}

 Can you please someone help us with the issue. 
 Thank you, 
 Varaprasad

**Comment by AbhijatSaxena** [comment, marked as answer]
Please enable "VNet Integration" in your logic app

Refer - https://techcommunity.microsoft.com/t5/azure-integration-services-blog/enabling-service-bus-and-sap-built-in-connectors-for-stateful/ba-p/3820381

We were facing similar issue with Complete Message & Dead letter message actions, it's resolved now.

**Reply by AbhijatSaxena** [reply]
"Service Bus and SAP built-in connectors in Logic App Standard contain many triggers and actions that were previously restricted to stateless Logic Apps only. Beginning in April, these triggers and actions are now available for use in stateful Logic Apps. This article explains the requirements needed to successfully use this new feature."
